### DISCUSSION TITLE
Some problems

### DISCUSSION THREAD

**Discussion by kamil-merdanov** [mclemente/fvtt-condition-lab-triggler#30] [Q&A] [answered]
Hello there and thank you for creating this module!

There are currently two things that I'm unable to bring into play:
1. Effects duration. It just doesn't work. I've tried different combinations. No matter how many rounds and turns pass effects are still there.
2. No matter what I paste/type into the Macro field it doesn't save. Clicking the save button doesn't do anything.

I'am using [AGE System (unofficial)](https://foundryvtt.com/packages/age-system) with a lot of other modules, but I've also tried your module only with [The Expanse RPG (AGE)](https://foundryvtt.com/packages/expanse). FoundryVTT version is 11.308.

**Comment by CalebKeas** [comment]
I'm also having the same problem with the effect macros, pushing the save button dose nothing.

**Reply by kamil-merdanov** [reply]
Does the effect duration work for you? I've tried it with DnD 5e. Still no reaction.

**Reply by CalebKeas** [reply]
Yes the duration does work for me, I used this video to make sure I had all the required modules (When it mentions "Combat Utility Belt" use "Condition Lab & Triggler") - https://www.youtube.com/watch?v=GzksEUs0v_c

**Reply by kamil-merdanov** [reply]
Thanks, I'll look into that. But it seems that the mentioned list of modules works only for dnd 5e.

**Comment by mclemente** [comment]
> 1. Effects duration. It just doesn't work. I've tried different combinations. No matter how many rounds and turns pass effects are still there.

Duration isn't handled by the module at all, it is meant to be handled by the system or another module that automates time.

> 2. No matter what I paste/type into the Macro field it doesn't save. Clicking the save button doesn't do anything.

Try toggling the "Remove Default Status Effects" setting on.

**Reply by kamil-merdanov** [reply]
> Duration isn't handled by the module at all, it is meant to be handled by the system or another module that automates time.

I see. It seems then, that it's not possible to use this feature with AGE/Expanse system. Does this also mean that Combat Triggers for Effect Macro won't work too?

> Try toggling the "Remove Default Status Effects" setting on.

It was already toggled on. The only time the "Save" button sort of works is when the text field is empty. It just closes the window.

![effect-macro](https://github.com/mclemente/fvtt-condition-lab-triggler/assets/157103190/d5efe585-ff03-4c73-b687-06ede8055fcf)

I'll keep digging. Any suggestions will be much appreciated. :)

**Reply by mclemente** [reply]
It means you need another module that handles time, like Simple Calendar or Small Time.

Effect Macro is its own module and I have no idea why it isn't working.

**Reply by kamil-merdanov** [reply]
I have simple calendar and it's tracks time increments (15 seconds per round) during combat. But it doesn't seem to help.
![simple-calendar](https://github.com/mclemente/fvtt-condition-lab-triggler/assets/157103190/5dab594a-2b4d-4f72-ba33-bf5baa64fd23)

**Reply by kamil-merdanov** [reply, marked as answer]
I've found a solution by chance: TImes up module by Tim Posney. You can close the discussion.
